src/clj/rems/logging.clj

```diff
@@ -0,0 +1,58 @@
+(ns rems.logging
+  (:require [clojure.test :refer [deftest is testing]])
+  (:import (org.slf4j MDC)))
+
+(defn with-mdc* [m f]
+  (let [m (->> m
+               (map (fn [[key val]]
+                      [(if (keyword? key)
+                         (name key)
+                         key)
+                       (str val)]))
+               (into {}))]
+    (try
+      (doseq [[key val] m]
+        (when-not (empty? val)
```

Comment:
If `val` is always a collection then `(seq val)` is correct.

In Clojure `k v` are usually preferred (also because `key` and `val` are both useful functions).